lib/features/children/parental_approval/dialogs/parental_approval_dialog.dart

```diff
@@ -0,0 +1,82 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:givt_app/app/injection/injection.dart';
+import 'package:givt_app/features/children/family_history/models/child_donation.dart';
+import 'package:givt_app/features/children/parental_approval/cubit/parental_approval_cubit.dart';
+import 'package:givt_app/features/children/parental_approval/widgets/parental_approval_approved_page.dart';
+import 'package:givt_app/features/children/parental_approval/widgets/parental_approval_confirmation_page.dart';
+import 'package:givt_app/features/children/parental_approval/widgets/parental_approval_declined_page.dart';
+import 'package:givt_app/features/children/parental_approval/widgets/parental_approval_error_page.dart';
+import 'package:givt_app/features/children/parental_approval/widgets/parental_approval_loading_page.dart';
+import 'package:go_router/go_router.dart';
+
+class ParentalApprovalDialog extends StatelessWidget {
+  const ParentalApprovalDialog({
+    required this.donation,
+    super.key,
+  });
+
+  final ChildDonation donation;
+
+  @override
+  Widget build(BuildContext context) {
+    final size = MediaQuery.sizeOf(context);
+    return BlocProvider(
+      create: (_) => ParentalApprovalCubit(
+        donation: donation,
+        decisionRepository: getIt(),
+      ),
+      child: Center(
+        child: SizedBox(
+          width: size.width * 0.75,
+          height: size.width * 0.75,
```

Comment:
Why based on width and not on height?

Reply:
yep, I just need square dialog with horizontal margins, so I calculate its size by screen width